### PR TITLE
Support enrich remote mode

### PR DESCRIPTION
**Pull request by dnhatn**
This PR supports enrich in remote mode. Enrich with remote mode can't be after another enrich with the coordinator mode, an aggregation, or a limit. While we can't address the first two limitations, we should remove the constraint with LIMIT. Otherwise, users are forced to write queries that may not perform well. For instance, 

```
FROM test
| ORDER @timestamp 
| LIMIT 10 | ENRICH[ccq.mode:remote] 
```
does not work. In such cases, users must rewrite it as 

```
FROM test 
| ENRICH[ccq.mode:remote]
| ORDER @timestamp 
| LIMIT 10
```

which is equivalent to bringing all data to the coordinating cluster.

We might consider implementing the actual remote enrich on the coordinating cluster (like remote field extracting), however, this requires retaining the originating cluster and restructing pages for routing, which might be complicated.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java [hunk 1]
```diff
@@ -351,4 +355,46 @@ private static Failure validateUnsignedLongNegation(Neg neg) {
         }
         return null;
     }
+
+    /**
+     * Ensure that no remote enrich is allowed after a reduction or an enrich with coordinator mode.
+     * <p>
+     * TODO:
+     * For Limit and TopN, we can insert the same node after the remote enrich (also needs to move projections around)
```
Comment (dnhatn) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:363]: I will work on a follow-up for this.

File: x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EnrichHelper.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.elasticsearch.xpack.esql.plan.logical.Enrich;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.test.ESTestCase.randomBoolean;
+
+public class EnrichHelper {
+
+    static String randomEnrichCommand(EnrichPolicy policy, String name, Enrich.Mode mode) {
```
Comment (costin) [inline x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EnrichHelper.java:20]: This is better suited for `org.elasticsearch.xpack.esql.EsqlTestUtils`.

**Review by costin — APPROVED** [review]
LGTM - Thanks Nhat for adding an extensive set of tests!

File: x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EnrichHelper.java (hunk 2 above)
Comment (dnhatn) [reply x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EnrichHelper.java]: ++ I've moved it to https://github.com/elastic/elasticsearch/pull/104993/commits/6027453b5765b2bf0127613139dd3613e8f63da4

**Comment by dnhatn** [comment]
Thanks, Costin!

**Comment by dnhatn** [comment]
Please see #105095 for the follow-up work related to this PR.

**Pull request merged**